app/components/combined_system_message/index.js

```diff
@@ -3,31 +3,72 @@
 
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
+import {createSelector} from 'reselect';
 
-import {getProfilesByIds, getProfilesByUsernames} from 'mattermost-redux/actions/users';
+import {getMissingProfilesByIds, getMissingProfilesByUsernames} from 'mattermost-redux/actions/users';
 import {Preferences} from 'mattermost-redux/constants';
 import {getBool, getTeammateNameDisplaySetting} from 'mattermost-redux/selectors/entities/preferences';
-import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
+import {getCurrentUser, getProfiles, getUsersByUsername} from 'mattermost-redux/selectors/entities/users';
 
 import CombinedSystemMessage from './combined_system_message';
 
-function mapStateToProps(state) {
-    const currentUser = getCurrentUser(state);
-    return {
-        currentUserId: currentUser.id,
-        currentUsername: currentUser.username,
-        showJoinLeave: getBool(state, Preferences.CATEGORY_ADVANCED_SETTINGS, Preferences.ADVANCED_FILTER_JOIN_LEAVE, true),
-        teammateNameDisplay: getTeammateNameDisplaySetting(state),
+function makeGetUserProfiles() {
```

Comment:
is there a reason why this isn't done in the redux lib?

Reply:
No reason at all.  I thought of this as well while working on webapp.  I'll submit PR to mattermost-redux and then will update commit of this PR.

Reply:
Done update with mattermost-redux